packages/service-library/src/servicelib/monitoring.py

```diff
@@ -58,6 +59,10 @@ async def middleware_handler(request: web.Request, handler):
                 resp_time,
                 resp.status,
             )
+        except asyncio.CancelledError as exc:
+            # python 3.8 cancellederror is not an Exception
+            resp = web.HTTPRequestTimeout(reason=str(exc))
+            raise
```

Comment:
this is an unusual setup: sets ``resp`` ->  reraises -> ``finally`` returns ``resp`` ... 

Recovering ``resp`` heavily depends on the caller
```python

from asyncio import CancelledError


def fail():
    raise CancelledError("failed")

def go():
    try:
        fail()
    except CancelledError as err:
        resp = 42
        raise
    finally:
        resp+=1
        return resp


try:
    resp = go()
finally:
    print(resp)
```
```
$ python test.py
$ 43
```

Add a note in the return to avoid that we move it outside of  the ``finally``

Reply:
hmm shall I rework it? I mean, I'm trying to fix something that was there already. and pylance does not like it at all. I will look into it again.

Reply:
and I had to move the return out of finally because of the linter.

Reply:
@pcrespov  please check again now

Reply:
looks more clear :-) thx!